src/components/network/line-layer.js

```diff
@@ -64,34 +65,57 @@ export const LineLoadingZone = {
 };
 
 export function getLineLoadingZoneOfSide(
+    line,
     limit,
+    activeFlow,
     intensity,
-    lineFlowAlertThreshold
+    lineFlowAlertThreshold,
+    voltagLevel
 ) {
-    if (limit === undefined || intensity === undefined || intensity === 0) {
+    if (
+        limit === undefined ||
+        (intensity === undefined && activeFlow === undefined) ||
+        (intensity === 0 && activeFlow === 0)
+    ) {
         return LineLoadingZone.UNKNOWN;
     } else {
-        let threshold = (lineFlowAlertThreshold * limit) / 100;
-        if (intensity > 0 && intensity < threshold) {
+        const vNom = voltagLevel.nominalVoltage;
+        const valIntensity =
+            intensity !== undefined
+                ? intensity
+                : (activeFlow * 1000) / (vNom * Math.sqrt(3) * DEFAULT_COSPHI);
+        const threshold = (lineFlowAlertThreshold * limit) / 100;
+        if (valIntensity > 0 && valIntensity < threshold) {
             return LineLoadingZone.SAFE;
-        } else if (intensity >= threshold && intensity < limit) {
+        } else if (valIntensity >= threshold && valIntensity < limit) {
             return LineLoadingZone.WARNING;
         } else {
             return LineLoadingZone.OVERLOAD;
         }
     }
 }
 
-export function getLineLoadingZone(line, lineFlowAlertThreshold) {
+export function getLineLoadingZone(
```

Comment:
i have a suggestion here , we can eliminate the second call to getLineLoadingZoneOfSide and uses a ternary operator to check the value of zone1 and zone2 and return the appropriate zone.
something like that:

```
export function getLineLoadingZone(
    line,
    lineFlowAlertThreshold,
    voltageLevel1,
    voltageLevel2
) {
    const zone1 = getLineLoadingZoneOfSide(
        line,
        line.permanentLimit1,
        line.p1,
        line.i1,
        lineFlowAlertThreshold,
        voltageLevel1
    );
    return zone1 >=
        getLineLoadingZoneOfSide(
            line,
            line.permanentLimit2,
            line.p2,
            line.i2,
            lineFlowAlertThreshold,
            voltageLevel2
        )
        ? zone1
        : LineLoadingZone.OVERLOAD;
}
```